### ISSUE TITLE
Library Scan Not Working on V2

### ISSUE THREAD

**Issue by zachg99** [HaveAGitGat/Tdarr#323]
I was running v1 of Tdarr just fine (last version was 1.3005) and have switched to v2_preview after the release on the 24th by creating a new docker container.

I setup the new v2 container, pointed it at my library and hit the scan link but nothing seemed to happen.  Then after a while I had 5 items in my library. I could click scan again and I had 9 items. I've tried re-scanning a few times and I've gotten up to 14 items in the library out of a little over 1k that I had previously.

Tdarr v2 Docker Install
Using the same Media folder as v1
Using the same PUID/PGID as v1

I can see the sub folders under /media in the browser. I've tried mapping to one of the sub folders vs the top level and that doesn't seem to make a difference.

Compose file: 
```version: '3'
services:

  tdarr_server:
    container_name: tdarr_server
    image: haveagitgat/tdarr:v2-preview
    privileged: true
    restart: unless-stopped
    environment:
      - PUID=${PUID} # default user id, defined in .env
      - PGID=${PGID} # default group id, defined in .env
      - TZ=${TZ} # timezone, defined in .env
      - serverIP=tdarr_server #using internal docker networking. This should at least work when the nodes are on the same docker compose as the server 
      - serverPort=8266
      - webUIPort=8265
    volumes:
      - ${ROOT}/tdarr/config:/app/configs # config files
      - ${ROOT}/tdarr/server:/app/server/Tdarr # config files
      - ${ROOT}/tdarr/cache:/temp
      - ${ROOT}/tdarr/testmedia:/home/Tdarr/testmedia
      - ${ROOT}/tdarr/scripts:/home/Tdarr/scripts
      - /volume1/video:/media # library
    ports:
      - 8265:8265 #Exposed to access webui externally
      - 8266:8266 #Exposed to allow external nodes to reach the server
    logging:
      options:
        max-size: "2m"
        max-file: "3"

  tdarr_node:
    container_name: tdarr_node
    image: haveagitgat/tdarr_node:v2-preview
    privileged: true
    restart: unless-stopped
    depends_on: 
      - tdarr_server
    devices:
      - /dev/dri:/dev/dri
    environment:
      - PUID=${PUID} # default user id, defined in .env
      - PGID=${PGID} # default group id, defined in .env
      - TZ=${TZ} # timezone, defined in .env
      - serverIP=tdarr_server #container name of the server
      - serverPort=8266
      - nodeID=TDARRNODE_1
      - nodeIP=tdarr_node #container name of the node
      - nodePort=9267 #not exposed via a "ports: " setting as the server/node communication is done on the internal docker network and can communicate on all ports
    volumes:
      - ${ROOT}/tdarr/config:/app/configs # config files
      - ${ROOT}/tdarr/testmedia:/home/Tdarr/testmedia
      - ${ROOT}/tdarr/scripts:/home/Tdarr/scripts
      - /volume1/video:/media # library
    logging:
      options:
        max-size: "2m"
        max-file: "3"
```

**Comment by zachg99** [comment]
Just to add some more information I see the following in the Server logs:

```
- Finished in 0.288s
[2021-01-25T14:18:59.552] [WARN] Tdarr_Server - Exit approved. Closing process.
[2021-01-25T14:18:59.552] [INFO] Tdarr_Server - Scanner fNCaiEK4p:Finished
TypeError: Cannot read property 'isDirectory' of undefined
    at /snapshot/tdarr_express_be/src/fileScanner/fileScanner.js:257:30
    at callback (/snapshot/tdarr_express_be/node_modules/graceful-fs/polyfills.js:295:20)
    at FSReqCallback.oncomplete (fs.js:175:21)
[2021-01-25T14:18:59.553] [FATAL] Tdarr_Server - fatal:TypeError: Cannot read property 'isDirectory' of undefined
[2021-01-25T14:18:59.553] [INFO] Tdarr_Server - Finished in 0.289s
[2021-01-25T14:18:59.553] [WARN] Tdarr_Server - Exit approved. Closing process.
[2021-01-25T14:18:59.553] [INFO] Tdarr_Server - Scanner fNCaiEK4p:Finished
[2021-01-25T14:18:59.553] [WARN] Tdarr_Server - Exit approved. Closing process.
[2021-01-25T14:18:59.559] [INFO] Tdarr_Server - Sending files for extraction
[2021-01-25T14:19:00.307] [INFO] Tdarr_Server - fNCaiEK4p File scanner exited
events.js:292
      throw er; // Unhandled 'error' event
      ^
Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at process.target.send (internal/child_process.js:708:16)
    at addFileToDB (/snapshot/tdarr_express_be/src/fileScanner/fileScannerThread.js:22:11)
    at extractData (/snapshot/tdarr_express_be/src/fileScanner/fileScannerThread.js:51:3)
    at runExtractions (/snapshot/tdarr_express_be/src/fileScanner/fileScannerThread.js:59:9)
    at async processArray (/snapshot/tdarr_express_be/src/fileScanner/fileScannerThread.js:75:7)
Emitted 'error' event on process instance at:
    at internal/child_process.js:712:35
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  code: 'ERR_IPC_CHANNEL_CLOSED'
}
[2021-01-25T14:19:14.707] [INFO] Tdarr_Server - Turning folder watch off for:/media
```

**Comment by HaveAGitGat** [comment]
This should be fixed in 2.00.01. You can try it using this method:
https://github.com/HaveAGitGat/Tdarr/issues/327#issuecomment-767516511

Please reopen if you still have the issue.

**Comment by HaveAGitGat** [comment]
On Docker just set the tag to 2.00.01.  Will move into v2-preview when some other fixes are confirmed.
